src/script/descriptor.h

```diff
@@ -63,7 +63,7 @@ struct Descriptor {
 };
 
 /** Parse a descriptor string. Included private keys are put in out. Returns nullptr if parsing fails. */
-std::unique_ptr<Descriptor> Parse(const std::string& descriptor, FlatSigningProvider& out);
+std::unique_ptr<Descriptor> Parse(const std::string& descriptor, FlatSigningProvider& out, bool require_checksum = false);
```

Comment:
Instead of adding the new argument with a default value, why not create a new function `ParseChecked`? From the call site it is more clear, especially since it's statically defined where checksum is needed.

Reply:
Or you could do `auto desc = Parse(desc_str, provider, /* require_checksum = */ true);`.

Reply:
Did the latter.